boards/sim/sim/sim/scripts/Make.defs

```diff
@@ -72,11 +72,16 @@ NM = $(CROSSDEV)nm
 OBJCOPY = $(CROSSDEV)objcopy
 OBJDUMP = $(CROSSDEV)objdump
 
+# Note: -fshort-wchar for the case where NuttX and the host OS have
+# differnt wchar_t. On Nuttx, it's uint16_t. On macOS, it's 32-bit.
 CFLAGS := $(ARCHWARNINGS) $(ARCHOPTIMIZATION) \
-   $(ARCHCPUFLAGS) $(ARCHINCLUDES) $(ARCHDEFINES) $(EXTRAFLAGS) -pipe
+   $(ARCHCPUFLAGS) $(ARCHINCLUDES) $(ARCHDEFINES) $(EXTRAFLAGS) -pipe \
+   -fshort-wchar
```

Comment:
should we put -fshort-wchar to ARCHCPUFLAGS/ARCHCPUFLAGSXX?

Reply:
for C and CXX, we can.
but how about CPPFLAGS?

Reply:
Does really CPPFLAGS require -fshort-wchar? it seem that CPPFLAGS is designed to not use any language specific flag.

Reply:
i don't know how CPPFLAGS is designed.
the majority of the current uses of PREPROCESS in the tree are for C files, it seems.

Reply:
PREPROCESS is most used to process nx template file to generate c source file, so template files aren't C files, but the generated files are.

Reply:
how to handle other arch? it may more simpler to typedef wchar_t in C same as in C++ without -fshort-wchar.

Reply:
`-fshort-wchar` should just work for the majority of archs.

for the rest of archs which is not using clang or gcc, i dunno. i guess they don't have `__WCHAR_TYPE__` either.
i don't know if the issue (wchar_t type mismatch between the compiler and nuttx) exists there either.